src/reporter/github.test.js

```diff
@@ -0,0 +1,519 @@
+import { createRequire } from 'module';
+
+import { expect } from 'chai';
+import nock from 'nock';
+
+import GitHub from './github.js';
+
+const require = createRequire(import.meta.url);
+
+describe('GitHub', function () {
+  this.timeout(5000);
+
+  let github;
+  const MANAGED_LABELS = require('./labels.json');
+
+  before(() => {
+    github = new GitHub('owner/repo');
+  });
+
+  describe('#initialize', () => {
+    const scopes = [];
+
+    before(async () => {
+      const existingLabels = MANAGED_LABELS.slice(0, -2);
+
+      nock('https://api.github.com')
+        .get('/repos/owner/repo/labels')
+        .reply(200, existingLabels);
+
+      const missingLabels = MANAGED_LABELS.slice(-2);
+
+      for (const label of missingLabels) {
+        scopes.push(nock('https://api.github.com')
+          .post('/repos/owner/repo/labels', body => body.name === label.name)
+          .reply(200, label));
+      }
+
+      await github.initialize();
+    });
+
+    after(nock.cleanAll);
+
+    it('should create missing labels', async () => {
+      scopes.forEach(scope => expect(scope.isDone()).to.be.true);
+    });
+  });
+
+  describe('#getRepositoryLabels', () => {
+    let scope;
+    let result;
+    let labels;
+
+    before(async () => {
+      labels = [{ name: 'bug' }, { name: 'enhancement' }];
+
+      scope = nock('https://api.github.com')
+        .get('/repos/owner/repo/labels')
+        .reply(200, labels);
+
+      result = await github.getRepositoryLabels();
```

Comment:
Since this is the tested method, shouldn't it be called in the test code rather than in the `before`? 🙂